<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>840</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3DViewer v1.0</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="GLWidget" name="openGLWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>640</width>
      <height>480</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="fileName">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>640</width>
      <height>20</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Panel</enum>
    </property>
    <property name="text">
     <string>no file selected</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_n_vertices">
    <property name="geometry">
     <rect>
      <x>665</x>
      <y>55</y>
      <width>165</width>
      <height>20</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="text">
     <string>number of vertices:</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcdVertices">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>75</y>
      <width>170</width>
      <height>25</height>
     </rect>
    </property>
    <property name="smallDecimalPoint">
     <bool>false</bool>
    </property>
    <property name="digitCount">
     <number>10</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="label_n_edges">
    <property name="geometry">
     <rect>
      <x>665</x>
      <y>105</y>
      <width>165</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>number of edges:</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="lcdEdges">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>125</y>
      <width>170</width>
      <height>25</height>
     </rect>
    </property>
    <property name="smallDecimalPoint">
     <bool>false</bool>
    </property>
    <property name="digitCount">
     <number>10</number>
    </property>
    <property name="segmentStyle">
     <enum>QLCDNumber::Filled</enum>
    </property>
    <property name="value" stdset="0">
     <double>0.000000000000000</double>
    </property>
   </widget>
   <widget class="QPushButton" name="loadFile">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>10</y>
      <width>170</width>
      <height>40</height>
     </rect>
    </property>
    <property name="text">
     <string>LOAD FILE</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>165</y>
      <width>130</width>
      <height>3</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_projection">
    <property name="geometry">
     <rect>
      <x>665</x>
      <y>180</y>
      <width>165</width>
      <height>20</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="text">
     <string>Type of projection:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="projectionCombo">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>200</y>
      <width>170</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>central</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>parallel</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="backgroundButton">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>240</y>
      <width>170</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Background color</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame_v">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>280</y>
      <width>170</width>
      <height>115</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_vertices">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>5</y>
       <width>170</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;vertices&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_v_style">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>25</y>
       <width>150</width>
       <height>20</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="text">
      <string>style:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="vertexCombo">
     <property name="geometry">
      <rect>
       <x>45</x>
       <y>25</y>
       <width>120</width>
       <height>23</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>none</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>circle</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>square</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_v_size">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>55</y>
       <width>150</width>
       <height>20</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="text">
      <string>size:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="vertexSpinner">
     <property name="geometry">
      <rect>
       <x>45</x>
       <y>55</y>
       <width>120</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>20</number>
     </property>
     <property name="value">
      <number>2</number>
     </property>
    </widget>
    <widget class="QPushButton" name="vertexButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>85</y>
       <width>150</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>vertices color</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_e">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>405</y>
      <width>170</width>
      <height>115</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_edges">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>5</y>
       <width>170</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;edges&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_e_style">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>25</y>
       <width>150</width>
       <height>20</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="text">
      <string>style:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="edgeCombo">
     <property name="geometry">
      <rect>
       <x>45</x>
       <y>25</y>
       <width>120</width>
       <height>23</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>solid</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>dashed</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_e_size">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>55</y>
       <width>150</width>
       <height>20</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="text">
      <string>thickness:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="edgeSpinner">
     <property name="geometry">
      <rect>
       <x>75</x>
       <y>55</y>
       <width>90</width>
       <height>23</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <widget class="QPushButton" name="edgeButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>85</y>
       <width>150</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>edges color</string>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_r">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>530</y>
      <width>260</width>
      <height>100</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_rotate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>5</y>
       <width>230</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;rotate&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_x">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>19</y>
       <width>20</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_y">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>44</y>
       <width>20</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_rotate_z">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>69</y>
       <width>20</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="rotateX">
     <property name="geometry">
      <rect>
       <x>25</x>
       <y>25</y>
       <width>215</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
    <widget class="QSlider" name="rotateY">
     <property name="geometry">
      <rect>
       <x>25</x>
       <y>50</y>
       <width>215</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
    <widget class="QSlider" name="rotateZ">
     <property name="geometry">
      <rect>
       <x>25</x>
       <y>75</y>
       <width>215</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>360</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_t">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>530</y>
      <width>260</width>
      <height>100</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_translate">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>5</y>
       <width>230</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;translate&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_translate_x">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>19</y>
       <width>20</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_translate_y">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>44</y>
       <width>20</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_translate_z">
     <property name="geometry">
      <rect>
       <x>5</x>
       <y>69</y>
       <width>20</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="translateX">
     <property name="geometry">
      <rect>
       <x>25</x>
       <y>25</y>
       <width>215</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-10</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
    <widget class="QSlider" name="translateY">
     <property name="geometry">
      <rect>
       <x>25</x>
       <y>50</y>
       <width>215</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-10</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
    <widget class="QSlider" name="translateZ">
     <property name="geometry">
      <rect>
       <x>25</x>
       <y>75</y>
       <width>215</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-10</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_z">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>530</y>
      <width>280</width>
      <height>30</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_zoom">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>5</y>
       <width>40</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;scale&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QSlider" name="scale">
     <property name="geometry">
      <rect>
       <x>55</x>
       <y>5</y>
       <width>195</width>
       <height>20</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="bmpButton">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>570</y>
      <width>89</width>
      <height>60</height>
     </rect>
    </property>
    <property name="text">
     <string>BMP</string>
    </property>
   </widget>
   <widget class="QPushButton" name="jpegButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>570</y>
      <width>89</width>
      <height>60</height>
     </rect>
    </property>
    <property name="text">
     <string>JPEG</string>
    </property>
   </widget>
   <widget class="QPushButton" name="gifButton">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>570</y>
      <width>100</width>
      <height>60</height>
     </rect>
    </property>
    <property name="text">
     <string>GIF</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
